src/components/TeacherComponents/ResourcePageLayout/ResourcePageLayout.tsx

```diff
@@ -67,144 +69,181 @@ const ResourcePageLayout: FC<ResourcePageLayoutProps> = (props) => {
+        </Flex>
+      </OakBox>
+
+      <Flex
+        $flexDirection="column"
+        $alignSelf="center"
+        $gap={16}
+        $maxWidth={420}
+      >
+        {props.showNoResources &&
+          (props.page === "download" ? (
+            <NoResourcesToDownload />
+          ) : (
+            <NoResourcesToShare />
+          ))}
+        {!props.showNoResources && (
+          <>
+            {props.showTermsAgreement && (
+              <TermsAgreementForm
+                form={{
+                  control: props.control,
+                  register: props.register,
+                  errors: props.errors,
+                  trigger: props.triggerForm,
+                }}
+                isLoading={props.showLoading}
+                email={props.email}
+                schoolId={props.schoolId}
+                schoolName={props.school}
+                setSchool={props.setSchool}
+                showSavedDetails={props.showSavedDetails}
+                handleEditDetailsCompletedClick={props.onEditClick}
+                showPostAlbCopyright={props.showPostAlbCopyright}
+                copyrightYear={props.updatedAt}
+              />
             )}
-          </Flex>
-        </OakFlex>
-      </OakFlex>
-    </Box>
+            {hasFormErrors && (
+              <OakFlex $flexDirection={"row"}>
+                <Icon name="content-guidance" $color={"red"} />
+                <OakFlex $flexDirection={"column"}>
+                  <OakP $ml="space-between-sssx" $color={"red"}>
+                    To complete correct the following:
+                  </OakP>
+                  <OakUL $mr="space-between-m">
+                    {getFormErrorMessages(props.errors).map((err, i) => {
+                      return (
+                        <OakLI $color={"red"} key={i}>
+                          {err}
+                        </OakLI>
+                      );
+                    })}
+                  </OakUL>
+                </OakFlex>
+              </OakFlex>
+            )}
+            {props.showTermsAgreement && props.cta}
+
+            {props.apiError && !hasFormErrors && (
+              <FieldError
+                id="download-error"
+                data-testid="download-error"
+                variant={"large"}
+                withoutMarginBottom
+                ariaLive="polite"
+              >
+                {props.apiError}
+              </FieldError>
+            )}
+          </>
+        )}
+      </Flex>
+    </OakFlex>
   );
-};
+}
 
-export default ResourcePageLayout;
+const DelayedLoadingSpinner = styled(OakLoadingSpinner)<{ $delay?: number }>`
+  ${({ $delay }) => {
+    if ($delay) {
+      return `
+        opacity: 0;
+        animation: delayed-spinner-show 0s;
+        animation-delay: ${$delay / 1000}s;
+        animation-fill-mode: forwards;
+      `;
+    }
+  }}
+
+  @keyframes delayed-spinner-show {
+    from {
+      opacity: 0;
+    }
+    to {
+      opacity: 1;
+    }
+  }
+`;
```

Comment:
I'm going to lift this out into oak-components in a follow up PR. It's quite common to want to delay the appearance of a loading spinner in the case of perceptibly instantaneous UI